### ISSUE TITLE
GPL => MIT License?

### ISSUE THREAD

**Issue by anvaka** [ProbableTrain/MapGenerator#29]
I was curious if you'd be open to switch GPL license to MIT?

I'd love to contribute to the project/build more stuff on top of it but I don't want to change all my open source projects to GPL license.

Note: I'm not a lawyer, and might be wrong here. I heard GPL requirements are much stricter when it comes to use/modifications: https://stackoverflow.com/questions/3902754/mit-vs-gpl-license so I always choose MIT in my projects

**Comment by doctorn** [comment]
As far as I'm aware, you'll only have to change your license if you include any of the source code from this project in future work. If you're just using it/contributing, it doesn't affect you.

**Comment by ProbableTrain** [comment]
I'm more tempted to keep GPL - from what I've read, the main difference between MIT and GPL is that under GPL, derivations cannot be distributed without releasing their source. In that sense, GPL could be seem as less strict from the user's perspective as distributed derivations must reveal the source.

From Stack Overflow:
>   If you want to restrict the use of your modifications, then MIT is able to be more restrictive than the GPL for distribution and that might be what you're looking for.
 >  In case you want to ensure that the freedom of your software does not get restricted that much by the users you distribute it to, then you might want to release under GPL instead of MIT.

But I'm open to discussion

**Comment by Salzian** [comment]
I don't really want to reopen an almost 2 year old issue but it fits the topic. 

I would like to use this technology within a game. Now the thing is, I would totally be willing to adapt the technology to a game engine like Unreal or Unity and make the plugin open-source, also crediting this repository. However, having to use the exact same license would hinder game developers to use it in their games as games are almost never open-source. The reasons to this are numerous. 

Now when I did some research, I stumbled upon GNU LGPLv3 (https://choosealicense.com/licenses/lgpl-3.0/). The only difference to the current license would be that developers can use the code / technology within a larger work without having to open source the entire work. If they do however make changes to the technology, they have to publish their modifications anyways.

So if someone would use this plugin in their game, they can publish the game without the full source. However, if they make modifications to the plugin, they would still have to publish these modifications to the plugin. To  me, GNU licenses are all about forcing bigger companies to give back to the community, which is great in my opinion. And GNU LGPL allows that without undermining their entire business model.

I'm not a fan of MIT myself as it's a bit too open for my taste. Maybe GNU LGPL would be a valid alternative to allow more use cases? If I would create a plugin for a game engine, I would obviously publish it under the same license as I wouldn't qualify that as a "larger work".
